### ISSUE TITLE
CMake build fails on macOS Sequoia with Xcode 16 release version

### ISSUE THREAD

**Issue by ADKaster** [swiftlang/swift-testing#723]
### Description

Cloning the repo with Xcode 16 selected and CMake 3.30.3 fails to build using the following steps:

```
cmake -B build -GNinja
ninja -C build
```

Xcode Version 16.0 (16A242d)

swift-driver version: 1.115 Apple Swift version 6.0 (swiftlang-6.0.0.9.10 clang-1600.0.26.2)
Target: arm64-apple-macosx15.0

cmake version 3.30.3

ninja version: 1.12.1

The build fails in the same way on both:

main: 47b127f1c98a9f4ff593288c7ee0c6618a912e6b
release/6.0.2: 3755346b1da0f611d3a8bd531831ac48c126b9d4

### Expected behavior

Same build as `swift build` is produced

### Actual behavior

Backtrace in the linked gist below

https://gist.github.com/ADKaster/8c62194d219a6f6599c2409d24e304e4

It appears to be a problem with swift-syntax?

```
1.	Apple Swift version 6.0 (swiftlang-6.0.0.9.10 clang-1600.0.26.2)
2.	Compiling with effective version 5.10
3.	While evaluating request ExecuteSILPipelineRequest(Run pipelines { Non-Diagnostic Mandatory Optimizations, Serialization, Rest of Onone } on SIL for TestingMacros)
4.	While running pass #396 SILFunctionTransform "OwnershipModelEliminator" on SILFunction "@$s11SwiftSyntax9TokenKindO2eeoiySbAC_ACtFZ".
 for '==(_:_:)' (in module 'SwiftSyntax')
 ```

### Steps to reproduce

git clone <swift-testing>
cd swift-testing
cmake -B build -GNinja
cmake --build build

### swift-testing version/commit hash

47b127f1c98a9f4ff593288c7ee0c6618a912e6b

### Swift & OS version (output of `swift --version ; uname -a`)

swift-driver version: 1.115 Apple Swift version 6.0 (swiftlang-6.0.0.9.10 clang-1600.0.26.2)
Target: arm64-apple-macosx15.0
Darwin Andrews-MacBook-Pro.local 24.0.0 Darwin Kernel Version 24.0.0: Mon Aug 12 20:52:12 PDT 2024; root:xnu-11215.1.10~2/RELEASE_ARM64_T6020 arm64

**Comment by ADKaster** [comment]
Also fails with Xcode 16.1 Beta 2

swift-driver version: 1.115 Apple Swift version 6.0 (swiftlang-6.0.0.9.11 clang-1600.0.26.2)
Target: arm64-apple-macosx15.0

Xcode Version 16.1 beta 2 (16B5014f)

**Comment by stmontgomery** [comment]
Do you get a different result if you first revert the change in #690?

**Comment by grynspan** [comment]
What's your thinking Stuart?

**Comment by ADKaster** [comment]
@stmontgomery Yes, applying this diff on top of 47b127f1c98a9f4ff593288c7ee0c6618a912e6b allows the build to complete 

```diff
diff --git a/Sources/TestingMacros/CMakeLists.txt b/Sources/TestingMacros/CMakeLists.txt
index c916204..bb55f09 100644
--- a/Sources/TestingMacros/CMakeLists.txt
+++ b/Sources/TestingMacros/CMakeLists.txt
@@ -31,8 +31,8 @@ if(SwiftTesting_BuildMacrosAsExecutables)
   set(FETCHCONTENT_BASE_DIR ${CMAKE_BINARY_DIR}/_d)
   FetchContent_Declare(SwiftSyntax
     GIT_REPOSITORY https://github.com/swiftlang/swift-syntax
-    GIT_TAG cb53fa1bd3219b0b23ded7dfdd3b2baff266fd25) # 600.0.0
-  FetchContent_MakeAvailable(SwiftSyntax)
+    GIT_TAG 27b74edd5de625d0e399869a5af08f1501af8837)
+FetchContent_MakeAvailable(SwiftSyntax)
 endif()
 ```

**Comment by grynspan** [comment]
Sounds like a compiler bug that's conflicting with the 600.0.0 tag. We aren't seeing the compiler issue with the latest main-branch toolchain (our CI is happy.) We should send this over to the Swift repo for triage, I think.

**Comment by ChaseKnowlden** [comment]
Failed to build for me as well.
